projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,645 @@
+/***************************************************************************//**
+ *   @file    ad2s1210_iio.c
+ *   @brief   Implementation of AD2S1210 IIO application interfaces
+********************************************************************************
+ * Copyright (c) 2023 Analog Devices, Inc.
+ * Copyright (c) 2023 BayLibre, SAS.
+ * All rights reserved.
+ *
+ * This software is proprietary to Analog Devices, Inc. and its licensors.
+ * By using this software you agree to the terms of the associated
+ * Analog Devices Software License Agreement.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+
+#include "iio.h"
+#include "iio_trigger.h"
+
+#include "ad2s1210_iio.h"
+#include "app_config.h"
+#include "ad2s1210_user_config.h"
+#include "common.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+
+/******** Forward declaration of getter/setter functions ********/
+static int iio_ad2s1210_attr_get(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+static int iio_ad2s1210_attr_set(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+/******************************************************************************/
+/********************* Macros and Constants Definition ************************/
+/******************************************************************************/
+#define AD2S1210_CHN_ATTR(_name, _priv) {\
+	.name = _name,\
+	.priv = _priv,\
+	.show = iio_ad2s1210_attr_get,\
+	.store = iio_ad2s1210_attr_set\
+}
+
+#define AD2S1210_CH(_name, _idx, _type, _ch_out) {\
+	.name = _name, \
+	.ch_type = _type,\
+	.ch_out = _ch_out,\
+	.indexed = true,\
+	.channel = 0,\
+	.scan_index = _idx,\
+	.scan_type = &chn_scan[_idx],\
+	.attributes = &ad2s1210_iio_ch_attributes[_idx][0]\
+}
+
+/* Number of IIO devices */
+#define NUM_OF_IIO_DEVICES	1
+
+#define	BYTES_PER_SAMPLE	2
+
+/* Number of data storage bits (needed for IIO client to plot RESOLVER data) */
+#define CHN_STORAGE_BITS	(BYTES_PER_SAMPLE * 8)
+
+#define AD2S1210_IIO_TRIGGER_NAME         "ad2s1210_iio_trigger"
+
+/* data buffer size */
+#if defined(USE_SDRAM)
+#define data_buffer				SDRAM_START_ADDRESS
+#define DATA_BUFFER_SIZE			SDRAM_SIZE_BYTES
+#else
+#define DATA_BUFFER_SIZE			(32768)		/* 32kbytes */
+static int8_t data_buffer[DATA_BUFFER_SIZE] = { 0 };
+#endif
+
+#define swab16(x) \
```

Comment:
Looks like a typo here (swap)

There's also this helper function in no-OS that can accomplish this by specifying a step size of 2

Reply:
not a typo, swab is the name used in  linux i had just picked the same name.
https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/swab.h
https://man7.org/linux/man-pages/man3/swab.3.html

that no os function looks more complex than just what this macro is doing, but ok
ill try it out and update the patch if it works

Reply:
> not a typo, swab is the name used in linux i had just picked the same name.

Oh I see, wasn't aware of this...